pyrasterframes/python/examples/Clustering.py

```diff
@@ -0,0 +1,88 @@
+
+#examples_setup
+from examples import resource_dir
+#examples_setup
+
+#py_cl_imports
+from pyrasterframes import *
+from pyrasterframes.rasterfunctions import *
+from pyspark.sql import *
+from pyspark.ml.feature import VectorAssembler
+from pyspark.ml.clustering import KMeans
+from pyspark.ml import Pipeline
+#py_cl_imports
+
+#py_cl_create_session
+spark = SparkSession.builder. \
+    master("local[*]"). \
+    appName("RasterFrames"). \
+    config("spark.ui.enabled", "false"). \
+    getOrCreate(). \
+    withRasterFrames()
+#py_cl_create_session
+
+# The first step is to load multiple bands of imagery and construct
+# a single RasterFrame from them.
+filenamePattern = "L8-B{}-Elkton-VA.tiff"
+bandNumbers = range(1, 8)
+bandColNames = list(map(lambda n: 'band_{}'.format(n), bandNumbers))
+
+# For each identified band, load the associated image file
+from functools import reduce
+joinedRF = reduce(lambda rf1, rf2: rf1.asRF().spatialJoin(rf2.asRF().drop('bounds').drop('metadata')),
```

Comment:
Curious to know if `rf2.asRF()` is required. If so, does it work if it's added after the `drop`?